bots/run-queue

```diff
@@ -56,41 +109,37 @@ def main():
     channel = connection.channel()
 
     try:
+        declare_webhook_result = channel.queue_declare(queue='webhook', passive=True, auto_delete=False)
         declare_public_result = channel.queue_declare(queue='public', passive=True, auto_delete=False)
         declare_rhel_result = channel.queue_declare(queue='rhel', passive=True, auto_delete=False)
     except pika.exceptions.ChannelClosed:
-        print("One of the queues doesn't exist")
         return 1
 
-    queue='public'
-    if redhat_network():
-        # Try the rhel queue if the public queue is empty
-        if declare_public_result.method.message_count == 0:
-            queue = 'rhel'
-        # If both are non-empty, shuffle
-        elif declare_rhel_result.method.message_count > 0:
-            queue = ['public', 'rhel'][random.randrange(2)]
+    if not any([declare_webhook_result.method.message_count,
+                declare_public_result.method.message_count,
+                declare_rhel_result.method.message_count]):
+        logging.warn("All queues are empty")
+        return 1
 
-    # Get one item from the queue if present
-    method_frame, header_frame, body = channel.basic_get(queue=queue)
-    if method_frame:
-        ret = 0
-        body = json.loads(body)
-        sys.stderr.write("Consuming {0} task:\n{1}\n".format(body['type'],json.dumps(body, indent=2, sort_keys=True)))
-        sys.stderr.flush()
-        p = subprocess.Popen(body['command'], shell=True)
-        while p.poll() is None:
-            connection.process_data_events()
-            time.sleep(3)
-
-        if p.returncode in [0, 2]:
-            channel.basic_ack(method_frame.delivery_tag)
+    if random.randrange(2):
+        cmd, body = consume_webhook_queue(channel, amqp)
     else:
-        # nothing in queue, no work done
-        ret = 1
+        cmd, body = consume_task_queue(channel, amqp, declare_public_result, declare_rhel_result)
+
+    if not cmd:
+        logging.warn("Unable to construct command")
+        return 1
```

Comment:
IMHO that should raise an exception, so that it's easier to spot in the logs.

Reply:
the most likely thing that would cause this is if another cockpit-task bot grabbed the message just in between declaring the queue and consuming it. Not sure if it's actually a failure, but it can throw an exception for standing out I guess.

Reply:
Err, how can that happen? The whole point of AMQP is to avoid such races..

Reply:
well declaring the queue != consuming. 

By declaring the queue you say you intend to consume from a queue, optionally creating it if `passive=False`. By declaring it you get some info back, how many messages are in the queue etc. there's a check on those numbers at the start. 

Then if any is >1 you try to consume from that queue. However if another bot in the meantime locked on a message (between declaring and consuming). You lose the message.

Typing this out I realize the first check is actually redundant.